### ISSUE TITLE
Break up the Spin `Configuration` page

### ISSUE THREAD

**Issue by itowlson** [fermyon/developer#215]
It currently combines:

* The Spin application manifest
* The component custom configuration system
* The Spin runtime configuration file

Having them all in together is quite confusing, at least to me.

Also, this is a broader issue but 
![image](https://user-images.githubusercontent.com/865538/204901662-1cc2f61e-35eb-40e5-8641-b4107690dae4.png)

**Comment by radu-matei** [comment]
I agree — I think we want a larger refresh for the Spin documentation, and this is a big item that we should tackle.
